instagram-app/src/components/authentication/withAuthenticate.jsx

```diff
@@ -0,0 +1,37 @@
+import React,{ Component } from "react";
+
+export default Comp => Auth =>(
+    class extends Component{
+        state={
+            loggedIn:false
+        }
+        componentDidMount(){
+            this.setState({loggedIn:localStorage.getItem('username')})
+        }
+
+
+
+        render(){
+            if(this.state.loggedIn)
+                return <Comp logout={this.logout}/>
+            else
+                return <Auth login={this.login}/>
+        }

```

Comment:
Great way to conditionally render the buttons for login and out here depending on if the user is logged in or not.